From c27c5cf56bec82d1d5c193a8fecb5a6da3f8be28 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Fri, 7 Sep 2018 10:14:57 +0300
Subject: [PATCH 4589/5242] MLK-19500: drm: imx: dcss: fix tracing function

commit  e2aa56b951db19aa476a90227f18492673e0c27d from
https://source.codeaurora.org/external/imx/linux-imx.git

kzalloc with GFP_KERNEL argument, may sleep. dcss_trace_write() is
called also from interrupt context and sleeping is not acceptable.

This patch will make kzalloc() call atomic.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-common.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dcss/dcss-common.c b/drivers/gpu/imx/dcss/dcss-common.c
index d8094a1..d4a9fe3 100644
--- a/drivers/gpu/imx/dcss/dcss-common.c
+++ b/drivers/gpu/imx/dcss/dcss-common.c
@@ -353,7 +353,7 @@ void dcss_trace_write(u64 tag)
 	if (!dcss_tracing)
 		return;
 
-	trace = kzalloc(sizeof(*trace), GFP_KERNEL);
+	trace = kzalloc(sizeof(*trace), GFP_ATOMIC);
 	if (!trace)
 		return;
 
-- 
1.7.9.5

